src/components/ConnectButton.tsx

```diff
@@ -0,0 +1,52 @@
+import { Button, Text } from '@chakra-ui/react';
+import { Trans } from '@lingui/macro';
+import { ConnectKitButton } from 'connectkit';
+import type { FunctionComponent } from 'react';
+
+import { MetaMaskIcon } from './icons';
+
+type CustomConnectButtonProps = {
+  handleOnClick?: (() => void) | undefined;
+  isConnected: boolean;
+  ensName?: string | undefined;
+  truncatedAddress?: string | undefined;
+};
+
+export const CustomConnectButton: FunctionComponent<
+  CustomConnectButtonProps
+> = ({ handleOnClick, isConnected, ensName, truncatedAddress }) => {
+  return (
+    <Button
+      variant="primary"
+      leftIcon={<MetaMaskIcon width={{ base: '0', sm: '1.3rem' }} />}
+      iconSpacing="0"
+      onClick={handleOnClick}
+      width={isConnected ? '55%' : '50%'}
+      height="2.5rem"
+      gap="0.5rem"
```

Comment:
Please use a theme value (i.e., `gap="4", or whatever the equivalent of 0.5 rem is).